src/store/split-data-reducer.js

```diff
@@ -0,0 +1,192 @@
+      };
+    }
+    if (action.splitType !== EQUALLY) {
+      console.log(state);
+      return {
+        ...state,
+        splitType: action.splitType,
+        splits: newSplitsWithEmptyValues,
+        splitsTotalAmount: getSum(newSplitsWithEmptyValues),
+      };
+    }
+    return {
+      ...state,
+      splitType: action.splitType,
+    };
+  }
+  /*-----------------------------------------------------*/
+  ////////// AMOUNT ENTERED //////////////
+  /*-----------------------------------------------------*/
+  if (action.type === "AMOUNT_ENTERED") {
+    if (action.splitType === EQUALLY) {
+      if (state.splits.length === 0) {
+        return {
+          ...state,
+          totalAmount: action.amount,
+          splitsTotalAmount: action.amount,
+        };
+      }
+
+      const newSplits = newSplitsforEqually(action.amount, state.splits);
+      const splitValues = newSplits.map((split) => split.value);
+
+      return {
+        ...state,
+        splits: newSplits,
+        totalAmount: action.amount,
+        splitsTotalAmount: getSum(splitValues),
+      };
+    }
+    return {
+      ...state,
+      totalAmount: action.amount,
+    };
+  }
+  /*-----------------------------------------------------*/
+  ////////// EXACT INPUT ///////////////////
+  /*-----------------------------------------------------*/
+  if (action.type === "EXACT_INPUT") {
+    const inputAmt = action.amount;
+    const newSplits = state.splits.map((split, i) => {
+      return {
+        ...split,
+        value: action.splitId === split.id ? inputAmt : split.value,
+      };
+    });
+    // console.log(newSplitsWithValues);
+    const splitValues = newSplits.map((split) => split.value);
+
+    return {
+      ...state,
+      splits: newSplits,
+      splitsTotalAmount: getSum(splitValues),
+    };
+  }
+
+  if (action.type === "NAME_CHANGE") {
+    const inputName = action.name;
+    const newSplitsWithNames = state.splits.map((split, i) => {
+      return {
+        ...split,
+        name: action.splitId === split.id ? inputName : split.name,
+      };
+    });
+
+    return {
+      ...state,
+      splits: newSplitsWithNames,
+    };
+  }
+  /*-----------------------------------------------------*/
+  ////////// ADD +++++++++ ///////////////////
+  /*-----------------------------------------------------*/
+  if (action.type === "ADD") {
+    const updatedSplits = state.splits.concat(newSplit);
+
+    if (action.splitType === EQUALLY) {
+      if (state.totalAmount === null)
+        return { ...state, splits: updatedSplits };
+
+      if (state.totalAmount != null) {
+        const newSplits = newSplitsforEqually(state.totalAmount, updatedSplits);
+
+        const splitValues = newSplits.map((split) => split.value);
+
+        return {
+          ...state,
+          splits: newSplits,
+          splitsTotalAmount: getSum(splitValues),
+        };
+      }
+      return {
+        ...state,
+        splits: updatedSplits,
+      };
```

Comment:
Your last return statement will never hit because the two `if` statements above check `state.totalAmount === null` and `state.totalAmount !== null`, returning at each, so there won't be a case your code gets past the second `if` statement. I would suggest cleaning up your code like this:

```jsx
if (state.totalAmount === null) {
  return { ...state, splits: updatedSplits };
}

const newSplits = newSplitsforEqually(state.totalAmount, updatedSplits);

const splitValues = newSplits.map((split) => split.value);

return {
  ...state,
  splits: newSplits,
  splitsTotalAmount: getSum(splitValues),
};
```

Here, you check once for `state.totalAmount === null`. If the execution logic gets past this `if` statement, then we **KNOW** that `state.totalAmount !== null` and we don't need to check it again.